### PR TITLE
Removed garbage_collect_data from __init__() functions

### LINKED ISSUE #90: Remove garbage_collect_data argument from all __init__ methods of pointers
At the moment, we want garbage collection to be always active when pointers to Doc, SubPipeline object are  lost. 

So each time there is a `garbage_collect_data` in any `__init__` function, this should be removed, in `DocPointer`, `SubPipelinePointer`, and others if exists

### PR DESCRIPTION
**Pull request by sachin-101**
## Description
Removes `garbage_collect_data` from all `__init__` methods of Pointers. 
Fixes #90 

## Type of Change
Please mark options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Documentation (non-breaking change which adds documentation)
- [x] Improvement (non-breaking change that improves the performance or reliability of existing functionality)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How has this been tested?
Existing tests

## Checklist
- [x] I did follow the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md)
- [x] I did follow the [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have added tests for my changes

### CODE REVIEW

**Comment by sachin-101** [comment]
@AlanAboudib It turns out that `.send()` implemented in `Base Worker` calls the object's `create_pointer` method passing `garbage_collect_data` as an argument. So we can't remove it as of now.

**Comment by AlanAboudib** [comment]
@sachin-101 I meant only removing it from _ _init_ _ methods, not from `create_pointer()`

File: syfertext/doc.py [hunk 1]
```diff
@@ -446,11 +445,7 @@ def create_pointer(
             owner = doc.owner
 
         doc_pointer = DocPointer(
-            location=location,
-            id_at_location=id_at_location,
-            owner=owner,
-            id=ptr_id,
-            garbage_collect_data=garbage_collect_data,
+            location=location, id_at_location=id_at_location, owner=owner, id=ptr_id
```
Comment (AlanAboudib) [inline syfertext/doc.py:448]: please add `garbage_collect_data = True`. Maybe I was not clear enough, excuse me. I wanted it to be removed only from the `__init__` function definition. Although, the super class have it as True by default, but I prefer we send it explicitely to super, to make sure that if PySyft guys changed the default, this wont' affect us

File: syfertext/pipeline/pointers/subpipeline_pointer.py [hunk 2]
```diff
@@ -28,29 +27,24 @@ def __init__(
...
         """
 
         # Initialize the parent object
         super(SubPipelinePointer, self).__init__(
-            location=location,
-            id_at_location=id_at_location,
-            owner=owner,
-            id=id,
-            garbage_collect_data=garbage_collect_data,
+            location=location, id_at_location=id_at_location, owner=owner, id=id
```
Comment (AlanAboudib) [inline syfertext/pipeline/pointers/subpipeline_pointer.py:37]: Same as above

File: syfertext/pipeline/subpipeline.py [hunk 3]
```diff
@@ -176,33 +175,28 @@ def create_pointer(
...
         """
 
         # Create the pointer object
         subpipeline_pointer = SubPipelinePointer(
-            location=location,
-            id_at_location=id_at_location,
-            owner=owner,
-            id=ptr_id,
-            garbage_collect_data=garbage_collect_data,
+            location=location, id_at_location=id_at_location, owner=owner, id=ptr_id
```
Comment (AlanAboudib) [inline syfertext/pipeline/subpipeline.py:192]: Same as above

File: syfertext/pointers/doc_pointer.py [hunk 4]
```diff
@@ -42,7 +41,6 @@ def __init__(
             id_at_location=id_at_location,
             owner=owner,
             id=id,
-            garbage_collect_data=garbage_collect_data,
```
Comment (AlanAboudib) [inline syfertext/pointers/doc_pointer.py:45]: same as above

File: syfertext/span.py [hunk 5]
```diff
@@ -290,11 +287,7 @@ def create_pointer(
             owner = span.owner
 
         span_pointer = SpanPointer(
-            location=location,
-            id_at_location=id_at_location,
-            owner=owner,
-            id=ptr_id,
-            garbage_collect_data=garbage_collect_data,
+            location=location, id_at_location=id_at_location, owner=owner, id=ptr_id
```
Comment (AlanAboudib) [inline syfertext/span.py:290]: same as above

**Comment by sachin-101** [comment]
@AlanAboudib I reverted back and added the new suggested changes. So some of your review comments would have vanished from the files changed section. :smile:

**Comment by AlanAboudib** [comment]
Any updates on this issue @sachin-101

**Comment by sachin-101** [comment]
> I agree on overriding the value in create_pointer methods
The overriding is done by passing `True` while initializing the Pointer Object. Hence the `garbage_collect_data` argument to the `Pointer object` constructor is independent of the value passed to the argument `garbage_collect_data` from PySyft.
And hence I haven't removed the argument from the constructor.

File: syfertext/doc.py (hunk 1 above)
Comment (sachin-101) [reply syfertext/doc.py:449]: The above is a constructor of `DocPointer` and if I add `garbage_collect_data = True` in it, then I can't remove it from ```__init__```, cause if I remove it from the constructor, I can't pass it while creating the object. :smile_cat:

**Pull request merged**